kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/fragment/MobiComConversationFragment.java

```diff
@@ -3896,14 +3902,14 @@ public void onClick(View v) {
                 emoticonsFrameLayout.setVisibility(View.GONE);
                 if (getActivity() != null) {
                     if (((KmStoragePermissionListener) getActivity()).isPermissionGranted()) {
-                        ((ConversationActivity) getActivity()).isTakePhoto(true);
+                        ((ConversationActivity) getActivity()).setFromCamera(true);
                         ((ConversationActivity) getActivity()).processCameraAction();
                     } else {
                         ((KmStoragePermissionListener) getActivity()).checkPermission(new KmStoragePermission() {
                             @Override
                             public void onAction(boolean didGrant) {
                                 if (didGrant) {
-                                    ((ConversationActivity) getActivity()).isTakePhoto(true);
+                                    ((ConversationActivity) getActivity()).setFromCamera(true);
                                     ((ConversationActivity) getActivity()).processCameraAction();
```

Comment:
@shubhamtewari is the method processCameraAction processing dependent on the value from fromCamera and that's the only place where 'fromCamera' is even used? means they both are dependent on each other and 'fromCamera' property doesn't have any significance without this method.
if yes, then it will be better to just pass the value of fromCamera as an argument to the function, without the need for setFromCamera method.

Reply:
the function itself isn't dependent on fromCamera. But, whether processCamera will be called or not, that is dependent on the fromCamera variable. 
permissions for the camera and video recording are common. so this variable decides wht will be called. processCamera or processVideoRecording.
Same is the case for processAttachment or processMultiselectGallery.

Reply:
removed the fromX variables

Reply:
using new requestCodess now